src/AggregateFunctions/QuantileGK.h

```diff
@@ -260,6 +260,10 @@ class ApproxSampler
 
         size_t sampled_len = 0;
         readBinaryLittleEndian(sampled_len, buf);
+        if (sampled_len > compress_threshold)
```

Comment:
If the user controls both `sampled_len` and `compress_threshold` this is not a sanity check. What if it sends 2^63 in both?

Reply:
Since it seems that users can't change the compress_threshold at all, we should check against that for both `compress_threshold` (which should be equal) and `sampled_len` (which should be <=). Am I understanding it correctly?

Reply:
Yes